elks/arch/i86/drivers/char/console-direct-pc98.c

```diff
@@ -48,27 +48,34 @@
 
 #define MAXPARMS	28
 
+#define MAX_OUTPUTS 1
+
 struct console;
 typedef struct console Console;
 
 struct console {
+    unsigned char op;
     int cx, cy;			/* cursor position */
     void (*fsm)(Console *, int);
     unsigned char attr;		/* current attribute */
     unsigned char XN;		/* delayed newline on column 80 */
-    unsigned int vseg;		/* video segment for page */
-    int basepage;		/* start of video ram */
+    unsigned int vseg;		/* vram for this console page */
+    int vseg_offset;		/* vram offset of vseg for this console page */
 #ifdef CONFIG_EMUL_ANSI
     int savex, savey;		/* saved cursor position */
     unsigned char *parmptr;	/* ptr to params */
     unsigned char params[MAXPARMS];	/* ANSI params */
 #endif
+    unsigned short w;
+    unsigned short max_col;
+    unsigned short h;
+    unsigned short max_row;
```

Comment:
One has to watch members declared after a member array declaration - these could increase instruction offset from byte to word, increasing code size. We're probably OK here, but give the params[] decl above, perhaps gate these variables and move them above CONFIG_EMUL_ANSI to be safe (MAXPARMS is fairy small I think).

Reply:
Fixed in 6ba7c8a8